packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -229,6 +232,10 @@ class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
   String get apksignerPath => getBuildToolsPath('apksigner');
 
   List<String> validateSdkWellFormed({bool requireApkSigner = true}) {
+    if (buildToolsVersion.major < minimumAndroidSdkVersion) {
```

Comment:
If you read the code in android_sdk.dart you'll see that we only select build tools versions that match the major version number of the platform tools. I'm happy to do the version check against both versions but flutter_tools requires platform version and build tools major version match